echopype/echodata/echodata.py

```diff
@@ -133,6 +142,118 @@ def __read_converted(self, converted_raw_path: Optional["PathHint"]):
 
         self.converted_raw_path = converted_raw_path
 
+    def _set_tree(self, tree: DataTree):
+        self._tree = tree
+
+    @classmethod
+    def from_file(cls, converted_raw_path, storage_options=None, open_kwargs={}):
+        echodata = cls(
+            converted_raw_path=converted_raw_path,
+            storage_options=storage_options,
+            open_kwargs=open_kwargs,
+        )
+        echodata._check_path(converted_raw_path)
+        converted_raw_path = echodata._sanitize_path(converted_raw_path)
+        suffix = echodata._check_suffix(converted_raw_path)
+        tree = open_datatree(
+            converted_raw_path,
+            engine=XARRAY_ENGINE_MAP[suffix],
+            **echodata.open_kwargs,
+        )
+        # Change root name to top
+        tree.name = "Top-level"
+
+        echodata._set_tree(tree)
+
+        if isinstance(converted_raw_path, fsspec.FSMap):
+            # Convert fsmap to Path so it can be used
+            # for retrieving the path strings
+            converted_raw_path = Path(converted_raw_path.root)
+        echodata.converted_raw_path = converted_raw_path
+        echodata._load_tree()
```

Comment:
It looks like `_load_tree()` is only necessary because we are currently trying to temporarily preserve the way we are accessing groups (e.g. ed.beam). Is this correct?

Reply:
Yes sir. `_load_tree` creates those dynamically generated attributes.

Reply:
Great!